### PR TITLE
Feature/424 separate graphs for train types

### LINKED ISSUE #424: Separate Graphs for Train Types
## User stories
As a user I want to see time window, verkehrsleistung und verkehrsmenge graphs for different transport types in order to see which transport types are used predominantly.

## Dependencies
Depends on: 
 - #344 

## (Wiki) References
- [Grafana Dashboard](https://github.com/BP2022-AP1/bp2022-ap1/wiki/Grafana-Dashboard)

## Acceptance Criteria
- [x] Run id verkehrsleistung coal, passenger
- [x] Run id verkehrsmenge coal, passenger
- [x] Config id verkehrsleistung coal, passenger
- [x] Config id verkehrsmenge coal, passenger
- [x] Config id window size
- [x] Multi config verkehrsleistung coal, passenger
- [x] Multi config verkehrsmenge coal, passenger
- [x] Multi config window size coal, passenger

### PR DESCRIPTION
**Pull request by LucasDerReisende**
Fixes #424 

Adapts and adds graphs to view differences in behavior of different train types

## PR checklist

- [x] Acceptance criteria fulfilled
- [x] Additional features are tested
- [x] Docs (code, wiki & diagrams) updated
- [x] Breaking changes anounced
- [x] Added classes that inherit from `BaseModel` to `src.constants.tables`
- [x] Dev-branch has been merged into local branch to resolve conflicts
- [x] Tests and linter have passed AFTER local merge
- [x] Another dev reviewed and approved

### CODE REVIEW

**Comment by LucasDerReisende** [comment]
## Updated Run Based Graphs
grafik

## Updated Config Based Graphs
grafik
grafik

## Updated Multi Config Based Graphs
grafik

File: src/data_science/data_science.py [hunk 1]
```diff
@@ -312,11 +312,30 @@ def get_verkehrsmenge_by_run_id(self, run_id: UUID) -> pd.DataFrame:
...
-                "verkehrsmenge": [block_section_times_df["block_section_length"].sum()],
+                "train_type": ["all"],
+                "verkehrsmenge": [grouped_df["verkehrsmenge"].sum()],
             }
         )
+        grouped_df = pd.concat(
+            [grouped_df, all_df],
+            ignore_index=True,
+        )
+        print('grouped_df', grouped_df)
```
Comment (antonykamp) [inline src/data_science/data_science.py]: Do you want to keep the print statement? 🙃

File: src/data_science/data_science.py [hunk 2]
```diff
@@ -713,6 +778,102 @@ def get_verkehrsleistung_by_config_id(self, config_id: UUID) -> pd.DataFrame:
...
+                    "block_section_length": data["block_section_length"].sum(),
+                }
+            )
+        )
+        all_df.reset_index(inplace=True)
+        grouped_df = pd.concat([grouped_df, all_df], axis=0)
+
+
+        grouped_df["verkehrsleistung"] = grouped_df.apply(
+            lambda row: row["block_section_length"] * 3600 / row["leave_tick"],
```
Comment (antonykamp) [inline src/data_science/data_science.py:869]: "3600" equals the conversion from seconds to hours, correct?

**Review by antonykamp — COMMENTED** [review]
Looks like a lot of work 😮 Thank you! I love that you were able to not use the train types as constants, but extract them. That makes your code very modular and extendable for new train types 🐝  And thanks for the screenshots 😇 
I just have two questions.

File: src/data_science/data_science.py (hunk 2 above)
Comment (LucasDerReisende) [reply src/data_science/data_science.py:869]: Yes, from km/s to km/h

**Review by antonykamp — APPROVED** [review]
Thanks for the clarification 💚

**Pull request merged**